### PR TITLE
Expand architecture docs

### PR DESCRIPTION
**Pull request by pianohacker**
Add a walkthrough of log/metric delivery from VM to consumer.

This is a DRAFT, comments highly appreciated.

### CODE REVIEW

File: architecture.html.md.erb [hunk 1]
```diff
@@ -8,7 +8,69 @@ This topic describes the log and metric egress architecture options. These archi
...
+
+<img src='images/architecture/overview.png' alt="Arrows are shown depicting the flow of information from VMs. One arrow points directly from VMs to Consumers. The other arrow points to the Firehose, which then sends to Consumers.">
+
+## <a id='vms-egress'></a> How Logs and Metrics Egress from VMs
+
+Logs and metrics are sent from VMs via the following process:
+
+<img src='images/architecture/vms.png' alt="A Forwarder Agent appears inside a square labeled VMs. Apps are shown sending to the Forwarder Agent, and Components are shown sending metrics to the the Forwarder Agent. Components also send logs to rsyslon, which then sends platform logs outside the system via syslog RFC5424. The Forwarder Agent sends to three downstream consumers. The Syslog Agent sends application logs via syslog RFC5424. The Metrics Agent exposes metrics via Prometheus endpoints. Finally, the Loggregator Agent sends metrics and application logs via syslog RFC5424.">
```
Comment (acrmp) [inline architecture.html.md.erb:19]: rsyslon should be rsyslog?

File: architecture.html.md.erb [hunk 2]
```diff
@@ -8,7 +8,69 @@ This topic describes the log and metric egress architecture options. These archi
...
+following process:
+
+<img src='images/architecture/firehose.png' alt="A Loggregator Agent appears inside a square labeled
+VMs. Loggregator Agents send to Dopplers, located on Doppler VMs. Dopplers, in turn, send to both
+Traffic Controllers and Reverse Log Proxies. V1 Firehose Consumers receive logs and metrics from
+Traffic Controllers, whereas V2 Firehose Consumers receive logs and metrics from Reverse Log
+Proxies. V2 Firehose Consumers can also receive logs and metrics from Reverse Log Proxy (RLP) Gateways.">
+
+  1. The Loggregator Agent takes each log/metric and sends it to one downstream Doppler
+     (distributing among a rondom subset of 5 Dopplers).
```
Comment (acrmp) [inline architecture.html.md.erb:42]: rondom should be random

File: architecture.html.md.erb [hunk 3]
```diff
@@ -83,11 +145,11 @@ The following components are included in the Shared-Nothing Architecture, as sho
...
 
-* **Forwarder Agent**: Forwarder Agents run on both <%= vars.platform_name %> component VMs and Diego Cell VMs. They receive logs and metrics from the apps and <%= vars.platform_name %> components located on those VMs. Forwarder Agents then forward the logs and metrics to Loggregator Agents and other agents.
+* **Forwarder Agent**: Forwarder Agents run on both <%= vars.platform_name %> component VMs and Diego Cell VMs. They receive logs and metrics from the applications and <%= vars.platform_name %> components located on those VMs. Forwarder Agents then forward the logs and metrics to Loggregator Agents and other agents.
 
 * **Syslog Agent**: <%= partial 'syslog_agents' %>
 
-* **Syslog Binding Cache** (not pictured): Syslog Agents can overwhelm CAPI when querying for existing bindings. This component acts a a caching proxy between the Syslog Agents and CAPI.
+* **Syslog Binding Cache** (not pictured): Syslog Agents can overwhelm CAPI when querying for existing bindings. This component acts as a caching proxy between the Syslog Agents and CAPI.
```
Comment (acrmp) [inline architecture.html.md.erb:152]: Maybe replace references to CAPI with "Cloud Controller API"?

Comment (pianohacker) [reply architecture.html.md.erb:152]: I'll leave this one up to the docs folks; not sure what standard terminology is here.

**Pull request merged**